### ISSUE TITLE
Cutycapt QtWebengine

### ISSUE THREAD

**Issue by RvdHout** [hoehrmann/CutyCapt#25]
Any plans or has anyone actually tried to convert CutyCapt to use (the new) QtWebengine instead of QtWebKit?

**Comment by Crystalix007** [comment]
See [Crystalix007/CutyCapt](https://github.com/Crystalix007/CutyCapt) for a version using the new QtWebEngine. Enough features needed to be removed / changed that I don't feel comfortable submitting a pull request, however, it does function for what I've tested it with.

**Comment by RvdHout** [comment]
I have build your version with QTCReator using msvc2017 (64), most options seem to work, only thing i noticed i can't output a PDF, can you confirm --out-format=pdf  doesn't work?

**Comment by Crystalix007** [comment]
Please see the latest commit. I was quitting too early (before the PDF ever had a chance to render), because I wasn't aware that the PDF rendering was asynchronous.

**Comment by RvdHout** [comment]
Yes that seems to work....i noticed you switched to printToPdf....the old method could not be used anymore? Asking for (optional/additional) papersize, orientation, quality & margin params

FYI,
--user-agent param seems to be ignored

**Comment by Crystalix007** [comment]
It would definitely work with the previous print method, but `printToPdf` also has a lot of configurability ([see Qt docs](https://doc.qt.io/qt-5/qwebenginepage.html#printToPdf)) as well. I just haven't added CLI options for them. Probably, because I don't personally need it, I'm not likely to implement it.

I don't see why useragent is being ignored. The processing is still there. Potentially I mucked something up when configuring the WebEngine object however.

**Comment by RvdHout** [comment]
I think --useragent needs QWebEngineProfile, not?
https://doc.qt.io/qt-5/qwebengineprofile.html#httpUserAgent

**Comment by Crystalix007** [comment]
Yep. Just fixed it so that a user agent is set in more than just the member variable of `CutyPage`.

**Comment by RvdHout** [comment]
Hi @Crystalix007, me again....i noticed some issues with the way you made pdf rendering work again, it seems you use --max-wait parameter for this, right?  Now lets say i set the --max-wait parameter like: --max-wait=10000 

Every PDF generated takes 10 seconds to complete... even if it is created much faster (very simple PDF made from a webpage with small table)
Now i can simply lower the --max-wait parameter, but that might interfere with other timeouts within cutycapt... is't there a better way to check PDF generation is complete?

**Comment by RvdHout** [comment]
`--max-wait=<ms>                    Don't wait more than (default: 90000, inf: 0)`

Btw, this also interferes with --max-wait=0 (infinite)

**Comment by Crystalix007** [comment]
Moved to my project's issues to avoid unnecessary notification noise for hoehrmann.
